### ISSUE TITLE
Speeding the dev det tracking

### ISSUE THREAD

**Issue by EmmaRenauld** [scilus/scilpy#697]
I know that that script is not really used, but: I just ran 10 000 seeds:

- Dipy det = 45 secs
- Dipy prob = 43 secs
- Our dev script dev = 22.5 minutes
- Out dev script prob = 3.5 minutes

We know that our dev script is snot made to be fast, but still... eventually we could investigate why the det is 7 times slower than prob.

**Comment by CHrlS98** [comment]
I think it's because it extracts the maximum directions inside the tracking cone at each step, which is way slower than simply sampling a distribution.

**Comment by grahamlittlephd** [comment]
https://github.com/scilus/scilpy/blob/2cc624f2deb501755e133c7be32b2e155d4f0836/scilpy/tracking/propagator.py#L535-L561

**Comment by CHrlS98** [comment]
Yes the for loop is the problem, but it is required to find local maximums and not global maximums. Maybe we can think of another way to get there, for instance by using dipy's `peak_directions` method or something?

I am not sure it behaves the same as Dipy's det tracking either, might be closer to EUDX tracking.

**Comment by arnaudbore** [comment]
Improve these lines from propagator.py in class ODFPropagator.

```
            for d in possible_maxima:
                new_cosinus = np.dot(v_in, d)
                if new_cosinus > cosinus:
                    cosinus = new_cosinus
                    v_out = d
```
